### PR TITLE
Add whitelist string and array policies, including SNS Raw Attribute Delivery

### PR DESCRIPTION
**Pull request by astuyve**
This (partially) closes https://github.com/mj1618/serverless-offline-sns/issues/41.
These changes only support whitelist string filtering (including OR matching), which seems to cover the reported use cases. If needed, we can revisit and support the full range of [supported filters](https://docs.aws.amazon.com/sns/latest/dg/sns-subscription-filter-policies.html)

It also skips the SNS envelope with RawMessageDelivery (for SQS->SNS delivery). Closing #55 in favor of this PR

### CODE REVIEW

File: src/types.d.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export interface ISNSAdapter {
     listSubscriptions(): Promise<ListSubscriptionsResponse>;
     unsubscribe(arn: string): Promise<void>;
     createTopic(topicName: string): Promise<CreateTopicResponse>;
-    subscribe(fnName: string, handler: SLSHandler, arn: string): Promise<void>;
+    subscribe(fnName: string, handler: SLSHandler, arn: string, policies: any): Promise<void>;
```
Comment (nicksieger) [inline src/types.d.ts]: Didn't see this signature, should change as follows:
```suggestion
    subscribe(fnName: string, handler: SLSHandler, arn: string, snsConfig: any): Promise<void>;
```

**Comment by astuyve** [comment]
Thanks @mj1618!

**Comment by astuyve** [comment]
@mj1618 do you know how long this will take to get merged and released? I'd like to set a reminder to update our local tooling to point back to a new version of Serverless Offline SNS, instead of my fork.

Thanks!

**Comment by mj1618** [comment]
Done @astuyve

**Comment by astuyve** [comment]
Thanks @mj1618!

**Comment by shierro** [comment]
Hi, i have this filter policy
```
    filterPolicy:
      operation:
      - "INSERT"
      newItemType:
      - prefix: "response"
```

and this is my code to publish sns data
```
/* eslint-disable no-console, import/no-extraneous-dependencies */
const AWS = require('aws-sdk');

const sns = new AWS.SNS({
  endpoint: 'http://127.0.0.1:4001',
  region: 'ap-southeast-2',
});

const stage = process.argv[2];

sns.publish(
  {
    Message: JSON.stringify({
      default: JSON.stringify({
        newItem: {
          ITEM_TYPE: 'response/onboarding/name',
          ITEM_VALUE:
            '{"S":"{\\"type\\":\\"name\\",\\"value\\":\\"shierro\\"}"}',
          ID: 'ResimaxVIP#1234',
        },
      }),
    }),
    MessageAttributes: {
      operation: {
        DataType: 'STRING_VALUE',
        StringValue: 'INSERT',
      },
      newItemType: {
        DataType: 'STRING_VALUE',
        StringValue: 'response',
      },
    },
    MessageStructure: 'json',
    TopicArn: `arn:aws:sns:ap-southeast-2:xxx:trm-conversation-${stage}`,
  },
  (err, result) => {
    console.log('err', err);
    console.log('new user published', result);
  },
);
```
but doesn't work. any advice on how to make the filter work offline?

**Comment by astuyve** [comment]
Hey @shierro - 
Right now prefix filters are not supported. We'd need to modify a fair bit of logic to support that.
Currently I'm not able to work on it, as I'm traveling for the next 30 days.

**Pull request merged**